pkcs7/src/signer_info.rs

```diff
@@ -0,0 +1,107 @@
+//! `SignerInfo` data type [RFC 5652 § 5.3](https://datatracker.ietf.org/doc/html/rfc5652#section-5.3)
+
+use core::cmp::Ordering;
+
+use crate::cms_version::CmsVersion;
+use der::{
+    asn1::{OctetStringRef, SetOfVec, UintRef},
+    Choice, Sequence, ValueOrd,
+};
+use spki::AlgorithmIdentifierRef;
+use x509_cert::{attr::Attribute, ext::pkix::SubjectKeyIdentifier, name::Name};
+
+/// ```text
+/// DigestAlgorithmIdentifier ::= AlgorithmIdentifier
+/// ```
+type DigestAlgorithmIdentifier<'a> = AlgorithmIdentifierRef<'a>;
+
+/// ```text
+/// SignatureAlgorithmIdentifier ::= AlgorithmIdentifier
+/// ```
+type SignatureAlgorithmIdentifier<'a> = AlgorithmIdentifierRef<'a>;
+
+/// ```text
+/// SignedAttributes ::= SET SIZE (1..MAX) OF Attribute
+/// ```
+type SignedAttributes<'a> = SetOfVec<Attribute>;
+
+/// ```text
+/// UnsignedAttributes ::= SET SIZE (1..MAX) OF Attribute
+/// ```
+type UnsignedAttributes<'a> = SetOfVec<Attribute>;
+
+/// ```text
+/// SignerIdentifier ::= CHOICE {
+//    issuerAndSerialNumber IssuerAndSerialNumber,
+//    subjectKeyIdentifier [0] SubjectKeyIdentifier }
+/// ```
+#[derive(Clone, Debug, PartialEq, Eq, Choice)]
+pub enum SignerIdentifier<'a> {
+    IssuerAndSerialNumber(IssuerAndSerialNumber<'a>),
+
+    #[asn1(context_specific = "0")]
+    SubjectKeyIdentifier(SubjectKeyIdentifier<'a>),
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Sequence)]
+#[allow(missing_docs)]
+pub struct IssuerAndSerialNumber<'a> {
+    pub name: Name,
+    pub serial_number: UintRef<'a>,
```

Comment:
Hmm, this is unfortunate. We have an `x509_cert::serial_number::SerialNumber` type, and I feel like this is what should be used here?